### ISSUE TITLE
Adding peer by bulk crash dashboard

### ISSUE THREAD

**Issue by zverstal** [donaldzou/WGDashboard#160]
Hello WGDashboard contributors,

When I try to add peer (even 1) by bulk everything is crash and wg-dashboard.service is go down.

![image](https://user-images.githubusercontent.com/63104825/159158034-91c4ff1a-c431-4ac5-a3bc-72db6af3cc2e.png)
<img width="294" alt="Снимок экрана 2022-03-20 142648" src="https://user-images.githubusercontent.com/63104825/159158148-b53579da-7780-4ef3-ad02-111061a181af.png">

**OS Information:**
 - OS:  Ubuntu 20.04.4
 - Python Version: 3.8.10

**Comment by donaldzou** [comment]
Hi, do you mind running the dashboard with `./wgd.sh debug` and try to add peers by bulk, to see that the error message is?

**Comment by zverstal** [comment]
<img width="444" alt="Снимок экрана 2022-03-20 231508" src="https://user-images.githubusercontent.com/63104825/159178956-66c5d83c-1786-4111-984f-07ec84d6fc92.png">

<img width="792" alt="Снимок экрана 2022-03-20 231737" src="https://user-images.githubusercontent.com/63104825/159179071-799179c0-5ea0-4ea0-8e31-fee61d2aa477.png">

Debug mod

**Comment by zverstal** [comment]
Also if I try to add **another DNS** to peer it's crash too with EMPTY RESPONSE ERROR too.

**Comment by donaldzou** [comment]
Hmm, could you please post the terminal output when you're adding DNS and add bulk peers?

**Comment by donaldzou** [comment]
Also I'm thinking, did you block your browser to set cookies? Because that's how the user info store.

**Comment by zverstal** [comment]
No proxy setting in my dashboard. I'm running it by default way (on my server external IP).

And i don't block any cookies. Just tried to add peers on Edge with default settings - result the same, dashboard is go down(

**Comment by donaldzou** [comment]
This is super weird... seems like your browser did not send the username and password successfully. Maybe did you have javascript disabled? or maybe your system is blocking it to fetch 3rd party libraries. This issue is kind of related to your issue: https://github.com/donaldzou/WGDashboard/issues/137#issuecomment-1020777272

**Comment by zverstal** [comment]
I haven't disabled Javascript, checked it in browser setting (i haven't changed any default settings in Chrome|Edge btw). I turned off any extensions and it's not help me either :(
And I don't have any blocklist on my Ubuntu... Even don't know how to set up it :)

**Comment by zverstal** [comment]
ipconfig /release.
ipconfig /flushdns.
ipconfig /renew.
netsh int ip set dns.
netsh winsock reset.

Delete browser history, cookies etc done too not working

**Comment by donaldzou** [comment]
This is weird... could you take a screen shot of your browser console output when you visit your dashboard?

**Comment by zverstal** [comment]
1. Log in, no errors, 200 status of JS components
<img width="956" alt="1" src="https://user-images.githubusercontent.com/63104825/160241469-a8d25d89-41af-4b1c-954d-9a3e71586dde.png">
2. Trying to add peers
<img width="960" alt="2" src="https://user-images.githubusercontent.com/63104825/160241485-47d1cc94-0684-406b-8227-11ecbeed626c.png">
3. Errors in JS
<img width="958" alt="3" src="https://user-images.githubusercontent.com/63104825/160241500-0d5fb1e5-035b-4b3a-a6ac-7d66ba2cd838.png">

**Comment by donaldzou** [comment]
This tells me that `net::ERR_EMPTY_RESPONSE` usually occur on the browser side... Do you mind try another browser, like Edge or Opera?

**Comment by zverstal** [comment]
Same situation in Edge and Opera (this one installed right now without any imports, just clean installation)

**Comment by zverstal** [comment]
I think there something in searching available IPs in database or something maybe?

**Comment by donaldzou** [comment]
I don't think so, because you can load peers. Just wondering, which version of dashboard you're using?

**Comment by zverstal** [comment]
3.0.6

**Comment by zverstal** [comment]
And yeah, i updated it from 2.3.8 and have new database. 

Also I have another server with clean installation of your dashboard (3.0.6), same errors in it :(

**Comment by donaldzou** [comment]
Hmmmmmm does your server exist locally? or is on the cloud?

**Comment by zverstal** [comment]
Cloud server.

**Comment by zverstal** [comment]
![image](https://user-images.githubusercontent.com/63104825/160244144-dace6071-a0be-4241-83d2-47c1f2ecd8c8.png)
Should I see here available IPs?

**Comment by donaldzou** [comment]
Yes, you should be. Do you mind tell me what `Address` you have for this configuration? the dashboard might crash when it is trying to generate too much IP.

**Comment by donaldzou** [comment]
the dashboard will generate available IP based on your configuration's address, if you have your subnet's mask less than `20`, it will take a long time to generate them.
![Screen Shot 2022-03-26 at 10 37 27 AM](https://user-images.githubusercontent.com/25237201/160244366-64f4e8d4-901c-47a2-9a1c-a42d301dd8f8.png)

**Comment by donaldzou** [comment]
> the dashboard will generate available IP based on your configuration's address, if you have your subnet's mask less than `20`, it will take a long time to generate them.

I forked this wonderful document on WireGuard: https://github.com/donaldzou/wireguard-docs

**Comment by zverstal** [comment]
I have 10.66.66.Х/32 for my configuration.

**Comment by zverstal** [comment]
https://github.com/angristan/wireguard-install

Maybe you know this popular installation script for Wireguard.
It's default network in this script 10.66.66.Х/32.

**Comment by donaldzou** [comment]
Ohhhhhh, you might use the wrong file lol. This script will generate 2 files (i think), one for **server**, one for **client**. The server's `.conf`'s `[Interface]` have `Address = xx.xx.xx.xx/24`, and your client's `.conf` will have `Address = xx.xx.xx.xx/32`

**Comment by donaldzou** [comment]
For a server, the address should have at least 1 available IP for the client, so if your server's configuration have `Address = xx.xx.xx.xx/32`, you should change it to `/24`, then it should be good, remember to switch off the configuration first. I'll look into the install script you used to see what is going on in there lol

**Comment by zverstal** [comment]
![image](https://user-images.githubusercontent.com/63104825/160244919-5c9c3298-adbd-4f5e-a91c-63f236260b4a.png)
from my wg0.conf 
it's 24 mask

**Comment by zverstal** [comment]
And peers have 32 mask, yeah

**Comment by donaldzou** [comment]
> ![image](https://user-images.githubusercontent.com/63104825/160244919-5c9c3298-adbd-4f5e-a91c-63f236260b4a.png) from my wg0.conf it's 24 mask

This is the first time i'm seeing 2 lines of `Address`, usually they exist all on the same line like `Address = 10.66.66.1/24, fd42:42:42::1/4`, can you switch `wg0` off with `wg-quick down wg0`, and remove the 2nd line of `Address` save it, and bring it back up to see if that solve everything lol. I know my dashboard did not designed on handle 2 lines of `Address`. I need to put that into my next release.

**Comment by zverstal** [comment]
It WORKS!!!
![image](https://user-images.githubusercontent.com/63104825/160245574-94117219-2546-4f44-be0e-46fe04830cf3.png)
B
Thank you a lot!
But problem here:
![image](https://user-images.githubusercontent.com/63104825/160245631-19a33a3c-93db-4988-be3c-f40e84b997c3.png)
this one is not working

**Comment by donaldzou** [comment]
Yessss!!! I do notice it does not handle IPv6 well. I will continue to look into that.

**Comment by zverstal** [comment]
Could I close this issue? For me it's solved :)

**Comment by donaldzou** [comment]
I'll close it ;) Have a great day/night!
